azkaban-exec-server/src/main/java/azkaban/execapp/JobRunner.java

```diff
@@ -210,28 +213,64 @@ private void createLogger() {
-      String logName = createLogFileName(node);
-      logFile = new File(workingDir, logName);
-
-      String absolutePath = logFile.getAbsolutePath();
+      // Attempt to create appenders as necessary
+      attachFileAppender();
 
-      jobAppender = null;
-      try {
-        RollingFileAppender fileAppender =
-            new RollingFileAppender(loggerLayout, absolutePath, true);
-        fileAppender.setMaxBackupIndex(jobLogBackupIndex);
-        fileAppender.setMaxFileSize(jobLogChunkSize);
-        jobAppender = fileAppender;
-        logger.addAppender(jobAppender);
-        logger.setAdditivity(false);
-      } catch (IOException e) {
-        flowLogger.error("Could not open log file in " + workingDir
-            + " for job " + this.jobId, e);
+      if (props.getBoolean("azkaban.job.kafkalogging.enabled", false)) {
+        attachKafkaAppender();
       }
     }
   }
 
+  private void attachFileAppender() {
+    jobAppender = null;
+    String logName = createLogFileName(node);
+    logFile = new File(workingDir, logName);
+    String absolutePath = logFile.getAbsolutePath();
+
+    try {
+      RollingFileAppender fileAppender =
+          new RollingFileAppender(loggerLayout, absolutePath, true);
+      fileAppender.setMaxBackupIndex(jobLogBackupIndex);
+      fileAppender.setMaxFileSize(jobLogChunkSize);
+      jobAppender = fileAppender;
+      logger.addAppender(jobAppender);
+      logger.setAdditivity(false);
+      flowLogger.info("Created file appender for job " + this.jobId);
+    } catch (IOException e) {
+      flowLogger.error("Could not open log file in " + workingDir
+          + " for job " + this.jobId, e);
+    }
+  }
+
+  private void attachKafkaAppender() {
+    kafkaAppender = null;
+
+    try {
+      KafkaLog4jAppender kafkaProducer = new KafkaLog4jAppender();
+      kafkaProducer.setBrokerList(props.getString("azkaban.job.kafkalogging.brokerlist"));
+      kafkaProducer.setTopic(props.getString("azkaban.job.kafkalogging.topic"));
+      kafkaProducer.setLayout(new EnhancedPatternLayout(
+          "{\"category\": \"%c{1}\","
+              + "\"level\": \"%p\","
+              + "\"message\": \"%m\","
+              + "\"projectname\": \"" + props.getString("azkaban.flow.projectname") + "\","
+              + "\"projectid\": \"" + props.getString("azkaban.flow.projectid") + "\","
+              + "\"flowid\": \"" + props.getString("azkaban.flow.flowid") + "\","
+              + "\"submituser\": \"" + props.getString("azkaban.flow.submituser") + "\","
+              + "\"execid\": \"" + props.getString("azkaban.flow.execid") + "\","
+              + "\"projectversion\": \"" + props.getString("azkaban.flow.projectversion") + "\""
+              + "}"));
+      kafkaProducer.activateOptions();
+      kafkaAppender = kafkaProducer;
+      logger.addAppender(kafkaAppender);
+      logger.setAdditivity(false);
+      flowLogger.info("Created kafka appender for " + this.jobId);
+    } catch (Exception e) {
```

Comment:
It could be possible to add more exception types.
However we need to keep using Exception as a catch-all since we want any and all failures to silently happen (we shouldn't crash the application due to the logger failing).